flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointOptions.java

```diff
@@ -47,7 +48,8 @@
         FORCED_ALIGNED
     }
 
-    public static final long NO_ALIGNMENT_TIME_OUT = Long.MAX_VALUE;
+    /** The maximum possible value is Long.MAX_VALUE 'ns' but API receives 'ms'. */
+    public static final long NO_ALIGNMENT_TIME_OUT = Duration.ofNanos(Long.MAX_VALUE).toMillis();
```

Comment:
What is the problem with the old value? After all, it's just a marker value. It does not have any true meaning.

Reply:
It was impossible to set NO_ALIGNMENT_TIME_OUT when alternating is enabled because when we are registering the register timer it converts this value to nanosecond which leads to the error because it was Long.MAX_VALUE.
We didn't see this problem because such a combination alternating + NO_ALIGNEMNT_TIME_OUT didn't meet before my changes but after I moved registerAlignmentTimer to checkNewCheckpoint some tests meet this problem.

Reply:
Hmm, ok I understand the problem, but then I think the solution is actually wrong. The `NO_ALIGNEMNT_TIME_OUT` means there should be no alignment timeout, therefore we should not register the timer.

We should rather add a condition in the `checkNewCheckpoint`:

```
   private void checkNewCheckpoint(CheckpointBarrier barrier) throws IOException {
        long barrierId = barrier.getId();
        if (currentCheckpointId < barrierId) {
            if (isCheckpointPending()) {
                cancelSubsumedCheckpoint(barrierId);
            }
            currentCheckpointId = barrierId;
            numBarriersReceived = 0;
            allBarriersReceivedFuture = new CompletableFuture<>();
            return true;
        if (currentCheckpointId >= barrierId) {
            return; // This barrier is not the first for this checkpoint.
        }

        if (isCheckpointPending()) {
            cancelSubsumedCheckpoint(barrierId);
        }
        currentCheckpointId = barrierId;
        numBarriersReceived = 0;
        allBarriersReceivedFuture = new CompletableFuture<>();
        firstBarrierArrivalTime = getClock().relativeTimeNanos();

        if (alternating && barrier.getCheckpointOptions().isTimeoutable()) {
            registerAlignmentTimer(barrier);
        }
        return false;
    }
```

Reply:
good point. I made such a fix